kernel/src/table_features/mod.rs

```diff
@@ -124,6 +150,424 @@ pub(crate) enum FeatureType {
+#[allow(dead_code)]
+#[derive(Clone)]
+pub(crate) enum KernelSupport {
+    /// Kernel has full support for any operation on this feature
+    Supported,
+    /// Kernel does not support this operation on this feature
+    NotSupported,
+    /// Custom logic to determine support based on operation type and table properties.
+    /// For example: Column Mapping may support Scan but not CDF, or CDF writes may only
+    /// be supported when AppendOnly is true.
+    Custom(fn(&Protocol, &TableProperties, Operation) -> DeltaResult<()>),
+}
+
+/// Types of requirements for feature dependencies
+#[allow(dead_code)]
+#[derive(Debug, Clone)]
+pub(crate) enum FeatureRequirement {
+    /// Feature must be supported (in protocol)
+    Supported(TableFeature),
+    /// Feature must be enabled (supported + property set)
+    Enabled(TableFeature),
+    /// Feature must NOT be supported
+    NotSupported(TableFeature),
+    /// Feature must NOT be enabled (may be supported but property must not activate it)
+    NotEnabled(TableFeature),
+    /// Custom validation logic
+    Custom(fn(&Protocol, &TableProperties) -> DeltaResult<()>),
+}
+
+/// Rich metadata about a table feature including version requirements, dependencies, and support status
+#[allow(dead_code)]
+#[derive(Clone)]
+pub(crate) struct FeatureInfo {
+    /// The feature's canonical name as it appears in the protocol
+    pub name: &'static str,
+    /// Minimum reader protocol version required for this feature
+    pub min_reader_version: i32,
+    /// Minimum writer protocol version required for this feature
+    pub min_writer_version: i32,
+    /// The type of feature (Writer, ReaderWriter, or Unknown)
+    pub feature_type: FeatureType,
+    /// Requirements this feature has (features + custom validations)
+    pub feature_requirements: &'static [FeatureRequirement],
+    /// Rust kernel's read support for this feature (may vary by Operation type)
+    pub read_support: KernelSupport,
+    /// Rust kernel's write support for this feature (may vary by Operation type)
+    pub write_support: KernelSupport,
+    /// How to check if this feature is enabled in a table
+    pub enablement_check: EnablementCheck,
+}
+
+// Static FeatureInfo instances for each table feature
+#[allow(dead_code)]
+static APPEND_ONLY_INFO: FeatureInfo = FeatureInfo {
+    name: "appendOnly",
+    min_reader_version: 1,
+    min_writer_version: 2,
+    feature_type: FeatureType::Writer,
+    feature_requirements: &[],
+    read_support: KernelSupport::NotSupported,
+    write_support: KernelSupport::Supported,
+    enablement_check: EnablementCheck::EnabledIf(|props| props.append_only == Some(true)),
+};
+
+#[allow(dead_code)]
+static INVARIANTS_INFO: FeatureInfo = FeatureInfo {
+    name: "invariants",
+    min_reader_version: 1,
+    min_writer_version: 2,
+    feature_type: FeatureType::Writer,
+    feature_requirements: &[],
+    read_support: KernelSupport::NotSupported,
```

Comment:
We should support reading tables that support Invariants. We just don't support them when we come across them.

If we didn't support, then we would not be able to read any tables between versions [2,6]

```suggestion
    read_support: KernelSupport::Supported,
```